src/persist-client/src/read.rs

```diff
@@ -687,34 +749,31 @@ where
     fn lease_seqno(&mut self) -> SeqNo {
         let seqno = self.machine.seqno();
 
-        *self.leased_seqnos.entry(seqno).or_insert(0) += 1;
+        *self
+            .leased_seqnos
+            .lock()
+            .expect("lock poisoned")
+            .entry(seqno)
+            .or_insert(0) += 1;
 
         seqno
     }
 
+    /// Returns a [`SubscriptionLeaseReturner`] tied to this [`ReadHandle`],
+    /// allowing a caller to return leases without needing to mutably borrowing
+    /// this `ReadHandle` directly.
+    pub(crate) fn clone_lease_returner(&self) -> SubscriptionLeaseReturner {
```

Comment:
nit: I usually prefer something like this to return a `&SubscriptionLeaseReturner` and then the `clone()` call can be moved up a few levels